string path
uint8 compression_method
uint8[] data